File: num3.java
import java.util.Scanner;

public class TimeChecker {
    public static boolean WhatTimeIsIt(int number) {
        int hours = number / 100;
        int minutes = number % 100;
        return (hours >= 0 && hours <= 23) && (minutes >= 0 && minutes <= 59);
    }

    public static void YesOrNo(int number, int n) {
        if (WhatTimeIsIt(number)) {
            System.out.println("YES");
        } else {
            System.out.println("NO");
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("n = ");
        int n = scanner.nextInt();
        for (int i = 0; i < n; i++) {
            System.out.print("Enter the number: ");
            int x = scanner.nextInt();
            YesOrNo(x, n);
        }
    }
}

File: num3.kt
import java.util.Scanner
fun whatTimeIsIt(number: Int): Boolean {
    val hours = number / 100
    val minutes = number % 100
    return (hours in 0..23) && (minutes in 0..59)
}

fun yesOrNo(number: Int, n: Int) {
    if (whatTimeIsIt(number)) {
        println("YES")
    } else {
        println("NO")
    }
}

fun main() {
  val scanner = Scanner(System.`in`)
    print("n = ")
    var n = 0
    n = scanner.nextInt()
    for (i in 0 until n) {
        print("Enter the number: ")
        var x = scanner.nextInt()
        yesOrNo(x, n)
    }
}
